foundation/blockchain/state/state.go

```diff
@@ -92,11 +91,7 @@ func New(cfg Config) (*State, error) {
 		return nil, err
 	}
 
-	storage, err := disk.New(cfg.DBPath)
-	//storage, err := memory.New()
-	if err != nil {
-		return nil, err
-	}
+	storage := cfg.Storage
```

Comment:
Don't do this, just add it to the struct underneath.

Reply:
Fixed.